### ISSUE TITLE
Prompt for update not working on iOS

### ISSUE THREAD

**Issue by phuze** [vite-pwa/vite-plugin-pwa#554]
When we release an update for our application, iOS user's (specifically iPads) don't receive a prompt to update.

This functionality works as expected outside of iOS (ie: Chrome on Windows).

The only way we can get the app to update on iPad, is by exiting/closing the app via the app drawer/app switcher. Upon launching the app again, it's immediately running the new version.

We're running vite `4.4.9` along with vite-plugin-pwa `0.16.4`.

Does iOS not support update prompts? Am I missing something necessary for this to function correctly on iOS?

```js
// vite.config.js

VitePWA({
  registerType: 'prompt',
  includeAssets: ['favicon.ico'],
  manifest: {
    name: 'My App',
    short_name: 'My App',
    description: 'My App',
    theme_color: '#BADA55',
    background_color: '#000000',
    display: 'standalone',
    orientation: 'landscape',
    icons: [ ... ]
  }
})
```

```js
// main.js

import { registerSW } from 'virtual:pwa-register'

const updateSW = registerSW({
  onNeedRefresh() {

    // note: we're using vue-sweetalert2 for dialog UI.
    // sweetalert works fine elsewhere in the app; so thats
    // not an issue here.
    Vue.swal({
      title: 'New Version',
      html: `New app version available. Reload?`,
      preConfirm: () => {
        return new Promise(resolve => {
          resolve()
        })
      }
    })
    .then(result => {
      if (result.value) {
        updateSW(true)
      }
    })
  }
})
```

**Comment by userquin** [comment]
@phuze 

Unfortunately macOS/iOS is a problem, it mainly depends on the version of OS and Safari. Trying to debug it is difficult, other than I don't have any macOS/iOS devices right now.

During the development of elk.zone we have encountered many problems on macOS/iOS (also in Firefox): for example, when installing the PWA, the keyboard was not displayed to compose a message. We also ran into the problem you describe, it seems that updating the version of macOS/iOS and Safari should fix it (Safari on iOS and iPadOS 16.4 b1 adds support for: https://twitter.com/firt/status/1626332953633603586?t=SIKmOJ8n1EtNDCSlaPMLzw&s= 33)

**Comment by phuze** [comment]
Thank you for the feedback. I can confirm I was running ipadOS `16.1` at the time of posting. I tried updating as you suggested -- the latest version being `16.6` in my case. Unfortunately, it had no effect.

Interestingly, returning to the iPad several hours after having pushed an update, I noticed that a prompt appeared alerting me that a new app version was available. This was surprising, but important, as it confirmed the update prompts technically function.

As I understand it, typically a waiting service worker will only become active once the currently active service worker is released and becomes redundant. From what I understand, this happens in one of two scenarios:
- A user navigates away from the page, thereby releasing the previous active worker.
- A period of time has passed, thereby releasing the previous active worker.

I tried enabling `skipWaiting` -- in Chrome on Windows, it just immediately activated the new service worker, rendering the update prompt meaningless. On the iPad, it effectively had no effect, since we weren't seeing update prompts to begin with.

```js
VitePWA({
  registerType: 'prompt',
  workbox: {
    skipWaiting: true
  },
  ...
})
```

With this in mind, I decided to implement an "update check" (a page reload) as part of route changes within my Vue app. I store  a timestamp for comparison, and trigger a page reload every 15 minutes, at the end of a route change. This provides an opportunity for any waiting service workers, and thus a user is presented with the update prompt.

I don't yet fully understand why iOS/ipadOS behaves the way it does, in comparison to Chrome in Windows or Android environment (which works flawlessly), but at least this method ensures app updates effectively reach our iPad users.

Here's an quick example demonstrating the approach I took within my app, incase anyone finds it helpful:

```js
// router
router.afterEach(() => {
  store.dispatch('checkUpdates')
})

// state
{
  lastUpdateCheck: null
}

// getter
lastUpdateCheck: state => {
  return state.lastUpdateCheck
}

// action
checkUpdates({ commit, getters }) {
  if(getters.lastUpdateCheck === null) {
    commit('setLastUpdateCheck')
  } else {
    if( ((Math.floor(Date.now() / 1000)) - getters.lastUpdateCheck) > 900) {
      commit('setLastUpdateCheck')
      window.location.reload()
    }
  }
}

// mutation
setLastUpdateCheck(state) {
  state.lastUpdateCheck = Math.floor(Date.now() / 1000)
}
```
